frontend/src/main/web/lib/constants/Views.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import Router from 'react-router';
+import StringUtils from '../utils/StringUtils';
+import RootContent from '../components/RootContent';
+import UserProfile from '../components/UserProfile';
+import SystemGlossary from '../components/SystemGlossary';
+
+var Views = {
+  USER_PROFILE: 0,
+  GLOSSARY: 1,
+
+  getView: function(value) {
+    if(StringUtils.isEmptyOrNull(value)) {
+      return null;
+    }
+
+    switch (value.toLowerCase()) {
+      case 'profile':
+      case 'user_profile':
+      case 'userProfile':
+        return this.USER_PROFILE; break;
+      case 'glossary': return this.GLOSSARY; break;
+      default: return null; break;
+    }
+  },
+
+  getRoutes: function(view) {
+    var Route = Router.Route,
+      DefaultRoute = Router.DefaultRoute,
+      NotFoundRoute = Router.NotFoundRoute;
+
+    if(StringUtils.isEmptyOrNull(view)) {
```

Comment:
This is being called with `view` as a `Number`, so it is misleading to use this (though it may work). If you change the view constants to use a `String` instead then this line is fine, otherwise it should be `_.isUndefined` or something.

On the other hand, is it even valid to call this with no argument? If not, consider an error instead so we can tell when we do something wrong.